nucypher/characters/lawful.py

```diff
@@ -1003,16 +1026,20 @@ def get_deployer(self):
         deployer = self._crypto_power.power_ups(TLSHostingPower).get_deployer(rest_app=self.rest_app, port=port)
         return deployer
 
+    @property
+    def decentralized_identity_evidence_from_metadata(self):
+        return self._metadata.payload.decentralized_identity_evidence or NOT_SIGNED
+
     def _generate_metadata(self) -> NodeMetadata:
         # Assuming that the attributes collected there do not change,
         # so we can cache the result of this method.
         # TODO: should this be a method of Teacher?
         timestamp = maya.now()
-        if self.decentralized_identity_evidence is NOT_SIGNED:
+        if self.federated_only:
             decentralized_identity_evidence = None
         else:
             decentralized_identity_evidence = self.decentralized_identity_evidence
-        payload = NodeMetadataPayload(canonical_address=self.canonical_address,
+        payload = NodeMetadataPayload(staker_address=self.canonical_address,
```

Comment:
To follow the convention this can be `staking_provider_address` everywhere in this PR.

Reply:
Ok, should I update the API & bump version on `nucypher-core` then?

Reply:
Heh, I suppose this is the meaning of integration of incoming changes with the rust-core!

Since the convention is so widespread I think it's worthwhile.

Reply:
Filed <https://github.com/nucypher/nucypher-core/issues/6>